enum.c

```diff
@@ -1385,6 +1392,180 @@ sort_by_cmp(const void *ap, const void *bp, void *data)
     return OPTIMIZED_CMP(a, b);
 }
 
+
+/*
+    This is parts of uniform sort
+*/
+
+#define uless rb_uniform_is_less
+
+static inline bool
+rb_uniform_is_less(VALUE a, VALUE b)
+{
+
+    if (FIXNUM_P(a) && FIXNUM_P(b)) {
+        return (SIGNED_VALUE)a < (SIGNED_VALUE)b;
+    }
+    else if (RB_FLOAT_TYPE_P(a) && RB_FLOAT_TYPE_P(b)) {
+        return rb_float_cmp(a, b) == -1;
+    }
+    else if (FIXNUM_P(a) && RB_FLOAT_TYPE_P(b)) {
+        return rb_float_cmp(b, a) == 1;
+    }
+    else if (RB_FLOAT_TYPE_P(a) && FIXNUM_P(b)) {
+        return rb_float_cmp(a, b) == -1;
+    }
+    else {
+        return rb_cmpint(rb_funcallv(a, id_cmp, 1, &b), a, b) == -1;
+    }
+}
+
+#define med3_val(a,b,c) (uless(a,b)?(uless(b,c)?b:uless(c,a)?a:c):(uless(c,b)?b:uless(a,c)?a:c))
+
+static void
+rb_uniform_insertionsort_2(VALUE *ptr_begin, VALUE* ptr_end)
+{
+    if (((ptr_end - ptr_begin)>>1) < 2) return;
+    for (VALUE *index = ptr_begin+2; index < ptr_end; index+=2) {
+        VALUE tmp = *index;
+        VALUE tmp2 = *(index+1);
+        VALUE *j = index, *k = index;
+        if (uless(tmp, *ptr_begin))
+            {
+                while (ptr_begin < j) {
+                    *j = *(k-=2);
+                    *(j+1) = *(k+1);
+                    j = k;
+                }
+            }
+            else {
+                while (uless(tmp, *(k-=2))) {
+                    *j = *k;
+                    *(j+1) = *(k+1);
+                    j = k;
+                }
+            }
```

Comment:
`if` and `else` aren't matched.

```suggestion
        if (uless(tmp, *ptr_begin)) {
            while (ptr_begin < j) {
                *j = *(k-=2);
                *(j+1) = *(k+1);
                j = k;
            }
        }
        else {
            while (uless(tmp, *(k-=2))) {
                *j = *k;
                *(j+1) = *(k+1);
                j = k;
            }
        }
```